go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -565,6 +565,408 @@ var DoltMerge = []enginetest.ScriptTest{
+				Expected: []sql.Row{{3}},
+			},
+			{
+				Query: "SELECT to_pk, to_c1, to_c2, from_pk, from_c1, from_c2, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit2 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{1, 2, 0, 1, 2, 3, "modified"},
+				},
+			},
+		},
+	},
+	{
+		Name: "base case: deleted row",
+		SetUpScript: []string{
+			"create table t (pk int primary key, c1 int, c2 int);",
+			"insert into t values (1, 2, 3), (4, 5, 6);",
+			"set @Commit1 = (select DOLT_COMMIT('-am', 'creating table t'));",
+
+			"delete from t where pk=1",
+			"set @Commit2 = (select DOLT_COMMIT('-am', 'modifying row'));",
+		},
+		Assertions: []enginetest.ScriptTestAssertion{
+			{
+				Query:    "SELECT COUNT(*) FROM DOLT_DIFF_t;",
+				Expected: []sql.Row{{3}},
+			},
+			{
+				Query: "SELECT to_pk, to_c1, to_c2, from_pk, from_c1, from_c2, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit2 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{nil, nil, nil, 1, 2, 3, "removed"},
+				},
+			},
+		},
+	},
+	{
+		// In this case, we do not expect to see the old/dropped table included in the dolt_diff_table output
+		Name: "table drop and recreate with overlapping schema",
+		SetUpScript: []string{
+			"create table t (pk int primary key, c int);",
+			"insert into t values (1, 2), (3, 4);",
+			"set @Commit1 = (select DOLT_COMMIT('-am', 'creating table t'));",
+
+			"drop table t;",
+			"set @Commit2 = (select DOLT_COMMIT('-am', 'dropping table t'));",
+
+			"create table t (pk int primary key, c int);",
+			"insert into t values (100, 200), (300, 400);",
+			"set @Commit3 = (select DOLT_COMMIT('-am', 'recreating table t'));",
+		},
+		Assertions: []enginetest.ScriptTestAssertion{
+			{
+				Query:    "SELECT COUNT(*) FROM DOLT_DIFF_t",
+				Expected: []sql.Row{{2}},
+			},
+			{
+				Query: "SELECT to_pk, to_c, from_pk, from_c, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit3 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{100, 200, nil, nil, "added"},
+					{300, 400, nil, nil, "added"},
+				},
+			},
+		},
+	},
+	{
+		// When a column is dropped we should see the column's value set to null in that commit
+		Name: "column drop",
+		SetUpScript: []string{
+			"create table t (pk int primary key, c1 int, c2 int);",
+			"insert into t values (1, 2, 3), (4, 5, 6);",
+			"set @Commit1 = (select DOLT_COMMIT('-am', 'creating table t'));",
+
+			"alter table t drop column c1;",
+			"set @Commit2 = (select DOLT_COMMIT('-am', 'dropping column c'));",
+		},
+		Assertions: []enginetest.ScriptTestAssertion{
+			{
+				Query:    "SELECT COUNT(*) FROM DOLT_DIFF_t;",
+				Expected: []sql.Row{{4}},
+			},
+			{
+				Query: "SELECT to_pk, to_c2, from_pk, from_c2 FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit1 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{1, 3, nil, nil},
+					{4, 6, nil, nil},
+				},
+			},
+			{
+				Query: "SELECT to_pk, to_c2, from_pk, from_c2 FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit2 ORDER BY to_pk;",
```

Comment:
I would expect this test case to select c1

Reply:
Right, see my other comment about AS OF for diff tables.

We want tests that demonstrate that dropped columns are shown as NULL in the diff table.